File: func_log/zadania1_5/zadanie5.js
// нефункциональный

function maxIdenticalImperative(numbers) {
    if (numbers.length === 0) return null;
    
    let maxValue = numbers[0];
    let maxLength = 1;
    let currentValue = numbers[0];
    let currentLength = 1;
    
    for (let i = 1; i < numbers.length; i++) {
        if (numbers[i] === currentValue) {
            currentLength++;
        } else {
            if (currentLength > maxLength) {
                maxLength = currentLength;
                maxValue = currentValue;
            }
            currentValue = numbers[i];
            currentLength = 1;
        }
    }
    
    // Проверка последней последовательности
    if (currentLength > maxLength) {
        maxLength = currentLength;
        maxValue = currentValue;
    }
    
    return maxValue;
}

// Пример использования
const numbers = [2, 4, 6, 2, 2];
console.log("Результат:", maxIdenticalImperative(numbers));  // 2

// функциональный

const maxIdenticalFunctionalGroupBy = numbers => {
    if (numbers.length === 0) return null;
    
    // Группируем последовательные одинаковые элементы
    const groups = numbers.reduce((acc, num) => {
        if (acc.length === 0 || acc[acc.length - 1].value !== num) {
            acc.push({ value: num, length: 1 });
        } else {
            acc[acc.length - 1].length++;
        }
        return acc;
    }, []);
    
    // Находим группу с максимальной длиной (первая среди равных)
    const maxGroup = groups.reduce((max, group) =>
        group.length > max.length ? group : max
    );
    
    return maxGroup.value;
};

// Пример использования
const numbers = [2, 4, 6, 2, 2];
console.log("Результат:", maxIdenticalFunctionalGroupBy(numbers));  // 2

File: func_log/zadania1_5/zadanie5.py
# нефункциональный

def max_identical_imperative(numbers):
    if not numbers:
        return None
    
    max_value = numbers[0]
    max_length = 1
    current_value = numbers[0]
    current_length = 1
    
    for i in range(1, len(numbers)):
        if numbers[i] == current_value:
            current_length += 1
        else:
            if current_length > max_length:
                max_length = current_length
                max_value = current_value
            current_value = numbers[i]
            current_length = 1
    
    # Проверка последней последовательности
    if current_length > max_length:
        max_length = current_length
        max_value = current_value
    
    return max_value


if __name__ == "__main__":
    data = input("Введите числа через пробел: ")
    numbers = [int(x) for x in data.split()]
    print(f"Входные данные: {numbers}")
    print(f"Максимальная идентичная последовательность: {max_identical_imperative(numbers)}")

    # функциональный

from itertools import groupby

def max_identical_functional_groupby(numbers):
    if not numbers:
        return None
    
    # Группируем одинаковые соседние значения
    groups = ((key, sum(1 for _ in group)) for key, group in groupby(numbers))
    
    # Находим группу с максимальной длиной (первая среди равных)
    max_group = max(groups, key=lambda x: x[1])
    
    return max_group[0]


if __name__ == "__main__":
    data = input("Введите числа через пробел: ")
    numbers = [int(x) for x in data.split()]
    print(f"Входные данные: {numbers}")
    print(f"Максимальная идентичная последовательность: {max_identical_functional_groupby(numbers)}")